nodejs/week3/Mealsharing/src/backend/api/meals.js

```diff
@@ -0,0 +1,202 @@
+const knex = require("../database");
+const express = require("express");
+const router = express.Router();
+//get method
+router.get("/meals", async (req, res) => {
+          try {
+                    const allMeals = await knex.from("meal").select("*");
+                    if (allMeals.length == 0) {
+                              res.status(404).send(
+                                        "Sorry data does not exists any more"
+                              );
+                    }
+                    res.send(allMeals);
+          } catch (err) {
+                    throw err;
+          }
+});
+//get method using id
+router.get("/meals/:id", async (req, res) => {
+          try {
+                    const mealId = req.params.id;
+                    const mealWithId = await knex
+                              .from("meal")
+                              .select("*")
+                              .where({ meal_id: mealId });
+                    if (mealWithId.length == 0) {
+                              res.status(404).send(
+                                        "Sorry data does not exists"
+                              );
+                    }
+                    res.send(mealWithId);
+          } catch (err) {
+                    throw err;
```

Comment:
how would you respond to the client if there's an error? What status code would be suitable?